src/components/structures/MatrixChat.js

```diff
@@ -891,6 +895,7 @@ module.exports = React.createClass({
         });
 
         cli.on('sync', function(state, prevState) {
+            dis.dispatch({action: 'sync_state', prevState, state});
```

Comment:
I don't really get why it's MatrixChat's job to relay this between the client and the store: if the store cares, shouldn't it just subscribe?